android/src/main/java/com/hypertrack/sdk/capacitor/CapacitorSerialization.kt

```diff
@@ -0,0 +1,138 @@
+        val exception = this.failure
+        if(exception is Exception) {
+            call.reject(exception.message, exception)
+        } else {
+            throw exception
+        }
+    }
+  }
+}
+
+@Suppress("UNCHECKED_CAST")
+internal fun List<Any>.toJSArray(): JSArray {
+    return JSArray().also { writableArray ->
+        forEach {
+            when (it) {
+                is String -> {
+                    writableArray.put(it)
+                }
+                is Map<*, *> -> {
+                    writableArray.put((it as Map<String, Any>).toJSObject())
+                }
+                else -> {
+                    throw Exception(IllegalArgumentException(it.javaClass.toString()))
+                }
+            }
+        }
+    }
+}
+
+@Suppress("UNCHECKED_CAST")
+internal fun Map<String, Any?>.toJSObject(): JSObject {
+    val map = this
+    return JSObject().apply {
+        map.forEach { entry ->
+            val key = entry.key
+            when (val value = entry.value) {
+                is String -> {
+                    put(key, value)
+                }
+                is Int -> {
+                    put(key, value)
+                }
+                is Double -> {
+                    put(key, value)
+                }
+                is Boolean -> {
+                    put(key, value)
+                }
+                is List<*> -> {
+                    put(key, (value as List<String>).toJSArray())
+                }
+                is Map<*, *> -> {
+                    put(key, (value as Map<String, Any>).toJSObject())
+                }
+                null -> {
+                    throw IllegalArgumentException("Invalid JSON: $this \n Null JSON values are not supported")
+                }
+                else -> {
+                    throw Exception(IllegalArgumentException(entry.value?.javaClass.toString()))
+                }
+            }
+        }
+    }
+}
+
+fun JSONObject.toMap(): Map<String, Any?> {
+    return keys().asSequence().associateWith { key ->
+        when(val value = this.get(key)) {
+            is Boolean,
+            is String,
+            is Double,
+            is Int -> {
+                value
+            }
+            is JSONArray -> {
+                value.toList()
+            }
+            is JSONObject -> {
+                value.toMap()
+            }
+            else -> {
+                null
+            }
+        }
+    }
+}
+
+private fun JSONArray.toList(): List<Any> {
+    return (0..length()).mapNotNull { index ->
+        when (val value = this.get(index)) {
+            is Boolean,
+            is String,
+            is Double,
+            is Int -> {
+                value
+            }
+            is JSONArray -> {
+                value.toList()
+            }
+            is JSONObject -> {
+                value.toMap()
+            }
+            else -> {
+                null
+            }
+        }
+    }
+}
```

Comment:
The format of the file is not up to kotlin standards. It mixes intend lengths. (2x space and 4x space)

Reply:
Fixed